server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -201,36 +209,97 @@ static ClusterState updateMetadata(String subscriptionName, ClusterState subscri
         }
     }
 
+    private static boolean updateIndexMetadataMapping(IndexMetadata.Builder updatedIndexMetadata,
+                                              IndexMetadata publisherIndexMetadata,
+                                              IndexMetadata subscriberIndexMetadata) {
+        var publisherMapping = publisherIndexMetadata.mapping();
+        var subscriberMapping = subscriberIndexMetadata.mapping();
+        if (publisherMapping != null && subscriberMapping != null) {
+            if (publisherIndexMetadata.getMappingVersion() > subscriberIndexMetadata.getMappingVersion()) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Updated index mapping {} for subscription {}", subscriberIndexMetadata.getIndex().getName(), publisherMapping.toString());
+                }
+                updatedIndexMetadata.putMapping(publisherMapping).mappingVersion(publisherIndexMetadata.getMappingVersion());
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static boolean updateIndexMetadataSettings(IndexMetadata.Builder updatedIndexMetadata,
+                                               IndexMetadata publisherMetadata,
+                                               IndexMetadata subscriberMetadata,
+                                               IndexScopedSettings indexScopedSettings) {
+        var isUpdated = false;
+        var publisherMetadataSetting = publisherMetadata.getSettings();
+        var subscriberMetadataSetting = subscriberMetadata.getSettings();
+        var newSubscriberIndexMetadataSettings = Settings.builder();
+        // Take all initial settings from the subscriber
+        for (var key : subscriberMetadataSetting.keySet()) {
+            newSubscriberIndexMetadataSettings.copy(key, subscriberMetadataSetting);
+        }
+        // Add or update all settings from the publisher. Skip private settings which are not transferable
+        // such as uuid or creation date and ignore static settings which are not allowed to change on published tables
+        for (var key : publisherMetadataSetting.keySet()) {
+            if (INDEX_NUMBER_OF_REPLICAS_SETTING.getKey().equals(key)) {
+                // Don't replicate number of replicas setting as this can get dynamically adapted
+                continue;
+            }
+            var setting = indexScopedSettings.get(key);
+            if (
+                setting != null &&
+                !setting.isInternalIndex() &&
+                !setting.isPrivateIndex() &&
+                indexScopedSettings.isDynamicSetting(key) &&
+                !indexScopedSettings.isPrivateSetting(key) &&
+                !Objects.equals(subscriberMetadataSetting.get(key), publisherMetadataSetting.get(key))
+            ) {
+                if (LOGGER.isDebugEnabled()) {
+                    LOGGER.debug("Update setting {} for index {}", key, publisherMetadata.getIndex());
```

Comment:
really minor as both indices should be the same but the settings are updated for `subscriberMetadata.getIndex()`